plugin-client/src/main/java/com/cloudera/nav/plugin/client/NavApiCient.java

```diff
@@ -146,17 +240,44 @@ private HttpHeaders getAuthHeaders() {
     return headers;
   }
 
-  private String getSourceUrl() {
+  /**
+   *
+   * @return url for querying all sources
+   */
+  private String getUrl() {
     // form the url string to request all entities with type equal to SOURCE
     String baseNavigatorUrl = config.getNavigatorUrl();
     String entities = joinUrlPath(baseNavigatorUrl, "entities");
-    return String.format("%s?query=%s", entities, SOURCE_QUERY);
+    return String.format("%s?query=%s%s", entities, SOURCE_QUERY, "&debugQuery=on");
+  }
+
+  /**
+   * @param type "entities", "relations"
+   * @param queryString result of getExtractorQueryString  made of desired extractorRunId's
+   * @return url for querying entities and relations
+   */
+  private String getUrl(String type, String queryString) {
+    String baseNavigatorUrl = config.getNavigatorUrl();
+    String typeUrl = joinUrlPath(baseNavigatorUrl, type);
+    return joinUrlPath(typeUrl, queryString);
   }
 
   private String joinUrlPath(String base, String component) {
     return base + (base.endsWith("/") ? "" : "/") + component;
   }
 
+  private String getExtractorQueryString(Map<String, Integer> m1, Map<String, Integer> m2){
+    HashSet<String> possibleExtractorRunIds = new HashSet<String>();
+    String queryString = "extractorRunId:(";
+    for (String key : m1.keySet()){
+      for (int i=m1.get(key); i==m2.get(key); i++){
+        String possible = key + "##" + Integer.toString(i);
```

Comment:
and you can use https://google-collections.googlecode.com/svn/trunk/javadoc/com/google/common/base/Joiner.html here